### ISSUE TITLE
use-unit-value

### ISSUE THREAD

**Issue by no-yan** [Monchi/eslint-plugin-chakra-ui#20]

**Comment by no-yan** [comment]
@Monchi what do you think?
And if you think it's worth implementing, do you think it's better to avoid mixing p={12} and w="12px"?
I mean, 

- Mixing
`<Box w="11px" h="12">`

- Not mixing
- `<Box w="11px" h="48px">`

**Comment by Monchi** [comment]
@no-yan Totally agree. I can't believe the second inconsistency...

> And if you think it's worth implementing, do you think it's better to avoid mixing p={12} and w="12px"?

I think we can prohibit number values since there is no merit except a very little reduction of bundle size.

Also I prefer the strict rule which forces the use of the single unit not only in the same element but in the whole application. How about  providing options like below?

```ts
type Option = {
  scope: "wholeApp" | "element",
  allowedUnits: ("px" | "rem")[],
}
```

**Comment by no-yan** [comment]
Thank you. I'm glad to hear it.

> Also I prefer the strict rule which forces the use of the single unit not only in the same element but in the whole application. How about providing options like below?
LGTM

Now let's implement it. I'll work on it after #21, but I don't mind if you do!

**Comment by Monchi** [comment]
@no-yan Ok, so please assign you to the issue when getting to work on it to avoid double work!

**Comment by no-yan** [comment]
I've made the prototype of the implementation, but I'm not sure which attribute is the one that receives the spacing value.

I will try to determine it by theme key from https://chakra-ui.com/docs/features/style-props. Column "Theme Key" seems to represent internal handling, and "space" correspond to "spacing" defined in theme.

If you have another idea, please let me know.

**Comment by Monchi** [comment]
@no-yan Maybe you can refer [this object](https://github.com/chakra-ui/chakra-ui/blob/c5eb9c03cd6925f8624ba48a406205d64000a2d1/packages/styled-system/src/config/space.ts#L5) but i'm not sure

**Comment by no-yan** [comment]
This does not handle width etc, so the other objects are also involved? I will look it closer tomorrow.

**Comment by Monchi** [comment]
@no-yan 

First, chakra calls `t.spaceT` for `margin` [here](https://github.com/chakra-ui/chakra-ui/blob/c5eb9c03cd6925f8624ba48a406205d64000a2d1/packages/styled-system/src/config/space.ts#L6:L6).

Then, `t.spaceT` calls `toConfig("space", transforms.px)` [here](https://github.com/chakra-ui/chakra-ui/blob/c5eb9c03cd6925f8624ba48a406205d64000a2d1/packages/styled-system/src/utils/index.ts#L17:L17).

`space` is `ThemeScale` that represents "Theme Key" in the doc, and `transforms.px` is a transformer to convert unitless value to px value.

You can see the same thing for `width` [here](https://github.com/chakra-ui/chakra-ui/blob/c5eb9c03cd6925f8624ba48a406205d64000a2d1/packages/styled-system/src/config/layout.ts#L6:L6).

So in summary, we can refer to `ThemeScale` like `space.margin.scale` or `layout.width.scale` to know which prop is treated as a spacing value.

**Comment by Monchi** [comment]
btw `width` is `t.sizesT` while `height` is `t.sizes`. This is the cause of your confusion 😂
